### PR TITLE
nit: Enables compiler warnings in part of loki module

### PR DESCRIPTION
**Pull request by PureW**

### CODE REVIEW

**Comment by PureW** [comment]
Osx errors out currently on the system-dependency prime_server:

```
In file included from /Users/distiller/project/valhalla/worker.h:12:
/usr/local/include/prime_server/http_protocol.hpp:124:18: error: 'prime_server::http_request_t::flush_stream' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
    virtual void flush_stream();
```

File: src/loki/worker.cc [hunk 1]
```diff
@@ -84,7 +84,7 @@ void loki_worker_t::parse_costing(Api& api, bool allow_none) {
   } catch (const std::runtime_error&) { throw valhalla_exception_t{125, "'" + costing_str + "'"}; }
 
   // See if we have avoids and take care of them
-  if (options.avoid_locations_size() > max_avoid_locations) {
+  if ((size_t)options.avoid_locations_size() > max_avoid_locations) {
```
Comment (mandeepsandhu) [inline src/loki/worker.cc:87]: Was this c-style cast suggested by clang? maybe do a `static_cast<size_t>` here?

Comment (PureW) [reply src/loki/worker.cc:87]: Oops, no suggestion, just an error about comparing different types. This is me being too used with C. I'll fix.

Comment (mandeepsandhu) [reply src/loki/worker.cc:87]: :+1:

File: src/loki/height_action.cc [hunk 2]
```diff
@@ -63,7 +63,7 @@ std::vector<PointLL> loki_worker_t::init_height(Api& request) {
   }
 
   // there are limits though
-  if (options.shape_size() > max_elevation_shape) {
+  if ((size_t)options.shape_size() > max_elevation_shape) {
```
Comment (mandeepsandhu) [inline src/loki/height_action.cc:66]: one more instance of the c-style cast

**Review by mandeepsandhu — APPROVED** [review]

**Review by mandeepsandhu — APPROVED** [review]

**Pull request merged**